pkg/agent/openflow/pipeline.go

```diff
@@ -2575,6 +2590,62 @@ func (c *client) createOFTable(table binding.Table, nextID uint8, missAction bin
 	return table
 }
 
+// igmpPktInFlows sends the IGMP packets to Antrea Agent on L3ForwardingTable.
+func (c *client) igmpPktInFlows(reason uint8) []binding.Flow {
+	pktInTable := L3ForwardingTable
+	flows := []binding.Flow{
+		// Send the IGMP packet to Antrea Agent and forward the packet normally in the OVS bridge if the packet is set
+		// with the "CustomReasonIGMPRegMark", so that the OVS multicast db cache can be updated, and Antrea Agent can
+		// identify the local multicast group and its members in the meanwhile.
+		pktInTable.BuildFlow(priorityHigh).
+			MatchProtocol(binding.ProtocolIGMP).
+			MatchRegMark(CustomReasonIGMPRegMark).
+			Action().SendToController(reason).
+			Action().Normal().Done(),
+	}
+	for _, dstIP := range []net.IP{mcastAllRouters, igmpV3ReportDst} {
+		// Set a custom reason for the IGMP packets in MulticastTable and send it to the pktInTable. "pktInTable" is
+		// only used to packet in the IGMP packet, and the other multicast traffic is forwarded normally or output to
+		// antrea-gw0 in the MulticastTable by the flows installed by "localMulticastForwardFlow" or
+		// "externalMulticastReceiverFlow".
+		flows = append(flows, MulticastTable.BuildFlow(priorityHigh).
```

Comment:
Why we process some multicast traffic in MulticastTable while processing others in L3ForwardingTable?

Reply:
When I'm reviewing https://github.com/antrea-io/antrea/pull/2964, I found networkpolicy packet-in handler for logging gets the custom reason from the register which is loaded by same OpenFlow that has the `SendToController` action. Why can't multicast handler gets the reason ion same way?
https://github.com/antrea-io/antrea/blob/486491482647c59157a22d68c6251474c22194e7/pkg/agent/openflow/pipeline.go#L1681-L1683

Reply:
updated.